libsupport/include/katana/Result.h

```diff
@@ -394,6 +400,28 @@ operator<<(std::ostream& out, const CopyableErrorInfo& ei) {
   return ei.Write(out);
 }
 
+}  // namespace katana
+
+BOOST_OUTCOME_V2_NAMESPACE_BEGIN
+
+namespace trait {
+
+template <>
+struct is_error_type<katana::ErrorInfo> {
+  static constexpr bool value = true;
+};
+
+template <>
+struct is_error_type<katana::CopyableErrorInfo> {
+  static constexpr bool value = true;
+};
+
+}  // namespace trait
+
+BOOST_OUTCOME_V2_NAMESPACE_END
```

Comment:
Can we move this after the katana namespace scope instead of the middle? Also can you verify and document that this is allowed? In general putting something in another libraries namespace is unsupported and unsafe.

Reply:
This is the specified way of adding a trait, see https://www.boost.org/doc/libs/1_70_0/libs/outcome/doc/html/reference/traits/is_error_type.html "Overridable: By template specialisation into the trait namespace."
I think these should be defined before the definition of the `Result` type to work, but I'm not sure if they would be also picked up if defined after the `Result`.